x/emissions/keeper/msgserver/msg_server_whitelist.go

```diff
@@ -16,20 +16,23 @@ func (ms msgServer) AddToWhitelistAdmin(ctx context.Context, msg *types.AddToWhi
 	if err != nil {
 		return nil, err
 	}
-	// Check that sender is also a whitelist admin
-	isAdmin, err := ms.k.IsWhitelistAdmin(ctx, msg.Sender)
+
+	// Validate the address
+	err = ms.k.ValidateStringIsBech32(msg.Address)
```

Comment:
We started this rough "defense-in-depth" practice where we purposefully do excessive data validations. Not the biggest fan of it because of the excess, however not doing this did cost us previously as we ended up storing "bad values" in the past. The theory is that functions could be reused in contexts without these checks, which could risk "poisoning" state as undesirable values are written to state.

If I were to remove the excessive validations, I would remove it from `whitelist.go`.

imo the best way to handle it is to wrap all primitive types in a domain language e.g. `type Address ...` instead of a `string` that is an address. Then when you construct the domain object, you do the validation once, and never have to again.

cc @relyt29 @xmariachi @spooktheducks @guilherme-brandao @fernandofcampos if they wanted to share thoughts

Reply:
Yes I see, I completely agree the type should ensure a _safety by design_, we can use `sdk.AccAddress` in the related `Keeper` method:
```go
func (k *Keeper) AddWhitelistAdmin(ctx context.Context, admin sdk.AccAddress)
```
Or even directly through the `ActorId` type:
```go
type ActorId = sdk.AccAddress
```
The last one provides a better expressiveness

This may bring a lot of changes, maybe better to address in a dedicated PR

Reply:
https://linear.app/alloralabs/issue/PROTO-2954/change-all-uses-of-primitive-types-to-use-classes

Great points. Saved for a follow-up ticket.